zio/jsonio/writer.go

```diff
@@ -1,26 +1,27 @@
 package jsonio
 
 import (
-	"encoding/json"
 	"io"
 
 	"github.com/brimdata/zed"
 )
 
 type Writer struct {
 	io.Closer
-	encoder *json.Encoder
+	encoder *encoder
```

Comment:
Is there a benefit to having the separate `encoder` type rather than just adding its fields and methods to `Writer`?

Reply:
Yeah we can do this.